Sources/SwiftIfConfig/ConfiguredRegions.swift

```diff
@@ -56,58 +143,111 @@ fileprivate class ConfiguredRegionVisitor<Configuration: BuildConfiguration>: Sy
   /// Whether we are currently within an active region.
   var inActiveRegion = true
 
+  /// Whether we are currently within an #if at all.
+  var inAnyIfConfig = false
+
+  // All diagnostics encountered along the way.
+  var diagnostics: [Diagnostic] = []
+
   init(configuration: Configuration) {
     self.configuration = configuration
     super.init(viewMode: .sourceAccurate)
   }
 
   override func visit(_ node: IfConfigDeclSyntax) -> SyntaxVisitorContinueKind {
-    // If we're in an active region, find the active clause. Otherwise,
-    // there isn't one.
-    let activeClause = inActiveRegion ? node.activeClause(in: configuration).clause : nil
+    // We are in an #if.
+    let priorInAnyIfConfig = inAnyIfConfig
+    inAnyIfConfig = true
+    defer {
+      inAnyIfConfig = priorInAnyIfConfig
+    }
+
+    // Walk through the clauses to find the active one.
     var foundActive = false
     var syntaxErrorsAllowed = false
+    let outerState: IfConfigRegionState = inActiveRegion ? .active : .inactive
     for clause in node.clauses {
-      // If we haven't found the active clause yet, syntax errors are allowed
-      // depending on this clause.
-      if !foundActive {
-        syntaxErrorsAllowed =
-          clause.condition.map {
-            IfConfigClauseSyntax.syntaxErrorsAllowed($0).syntaxErrorsAllowed
-          } ?? false
-      }
+      let isActive: Bool
+      if let condition = clause.condition {
+        if !foundActive {
+          // Fold operators so we can evaluate this #if condition.
+          let (foldedCondition, foldDiagnostics) = IfConfigClauseSyntax.foldOperators(condition)
+          diagnostics.append(contentsOf: foldDiagnostics)
```

Comment:
This is now duplicating logic from `IfConfigDeclSyntax.activeClause(in:)`, right? Do you think there’s any way to avoid the duplication? I’m a little worried that they will get out of sync eventually.

Reply:
We can do a little bit better here, because there were 4 copies of code like this.